main/src/library/List.flix

```diff
@@ -223,33 +223,47 @@ namespace List {
     ///
     /// Alias for `findLeft`.
     ///
-    /// The function `f` must be pure.
-    ///
     @Time(time(f) * length(l)) @Space(space(f))
-    pub def find(f: a -> Bool, l: List[a]): Option[a] = findLeft(f, l)
+    pub def find(f: a -> Bool & ef, l: List[a]): Option[a] & ef = findLeft(f, l)
 
     ///
     /// Optionally returns the first element of `l` that satisfies the predicate `f` when searching from left to right.
     ///
-    /// The function `f` must be pure.
-    ///
     @Time(time(f) * length(l)) @Space(space(f))
-    pub def findLeft(f: a -> Bool, l: List[a]): Option[a] = match l {
+    pub def findLeft(f: a -> Bool & ef, l: List[a]): Option[a] & ef = match l {
         case Nil => None
         case x :: xs => if (f(x)) Some(x) else findLeft(f, xs)
     }
 
     ///
     /// Optionally returns the first element of `l` that satisfies the predicate `f` when searching from right to left.
     ///
-    /// The function `f` must be pure.
-    ///
     @Time(time(f) * length(l)) @Space(space(f))
-    pub def findRight(f: a -> Bool, l: List[a]): Option[a] = match l {
+    pub def findRight(f: a -> Bool & ef, l: List[a]): Option[a] & ef =
```

Comment:
Whats the purpose of this change? Was the old code not fine (other than not being in CPS?)

Reply:
`f` was not effect polymorphic. Had to be pure